encryptonize.go

```diff
@@ -86,76 +90,99 @@ func New(keyProvider key.Provider) (Encryptonize, error) {
 	access.AddGroups(user.ID)
-	sealedAccess, err := access.Seal(sealedObject.ID, e.accessCryptor)
+	sealedAccess, err := access.Seal(oid, e.accessCryptor)
 	if err != nil {
-		return data.SealedObject{}, data.SealedAccess{}, err
+		return uuid.Nil, err
 	}
 
-	return sealedObject, sealedAccess, nil
+	// Write data to IO Provider
+	if err := e.putSealedObject(&sealedObject); err != nil {
+		return uuid.Nil, err
+	}
+	if err := e.putSealedAccess(&sealedAccess); err != nil {
+		return uuid.Nil, err
+	}
+
+	return oid, nil
 }
 
 // Update creates a new sealed object containing the provided plaintext data but uses a previously
-// created access object to control access to that data. The authorizing user must be part of the
-// provided access object, either directly or through group membership. The access object is
-// modified in-place, and any object previously associated with this access object can no longer be
-// decrypted.
-//
-// The sealed object is not sensitive data.
-func (e *Encryptonize) Update(authorizer *data.SealedUser, object *data.Object, access *data.SealedAccess) (data.SealedObject, error) {
+// created access list to control access to that data. The authorizing user must be part of the
+// provided access list, either directly or through group membership.
+func (e *Encryptonize) Update(authorizer *data.SealedUser, oid uuid.UUID, object *data.Object) error {
+	access, err := e.getSealedAccess(oid)
+	if err != nil {
+		return err
+	}
+
 	plainAccess, err := e.authorizeAccess(authorizer, access)
 	if err != nil {
-		return data.SealedObject{}, err
+		return err
 	}
 
-	wrappedOEK, sealedObject, err := object.Seal(access.ID, e.objectCryptor)
+	wrappedOEK, sealedObject, err := object.Seal(oid, e.objectCryptor)
 	if err != nil {
-		return data.SealedObject{}, err
+		return err
 	}
 
 	plainAccess.WrappedOEK = wrappedOEK
-	sealedAccess, err := plainAccess.Seal(sealedObject.ID, e.accessCryptor)
+	sealedAccess, err := plainAccess.Seal(oid, e.accessCryptor)
 	if err != nil {
-		return data.SealedObject{}, err
+		return err
 	}
-	*access = sealedAccess
 
-	return sealedObject, nil
+	// Write data to IO Provider
+	if err := e.putSealedObject(&sealedObject); err != nil {
+		return err
+	}
+	if err := e.putSealedAccess(&sealedAccess); err != nil {
+		return err
+	}
+
+	return nil
 }
 
-// Decrypt extracts the plaintext from a sealed object. The authorizing user must be part of the
-// provided access object, either directly or through group membership.
+// Decrypt fetches a sealed object and extracts the plaintext. The authorizing user must be part of
+// the provided access list, either directly or through group membership.
 //
 // The unsealed object may contain sensitive data.
-func (e *Encryptonize) Decrypt(authorizer *data.SealedUser, object *data.SealedObject, access *data.SealedAccess) (data.Object, error) {
+func (e *Encryptonize) Decrypt(authorizer *data.SealedUser, oid uuid.UUID) (data.Object, error) {
```

Comment:
What is the input UUID? I think all the documentation comments where we have an oid as input/output should be updated to be more clear that the oid is the IO provider identifier.